### PR TITLE
feat:JRuby 9.2 support

### PR DESCRIPTION
**Pull request by marcotc**
This PR adds **full support for JRuby 9.2.x** to `ddtrace`.
It finishes the work started in #995.

The only two integrations not supported are the ones not supported by JRuby itself: `grpc`, as [the official `protobuf` gem doesn't support JRuby currently](https://github.com/protocolbuffers/protobuf/issues/1594), and `mysql2`, as [JRuby uses JDBC directly for its database needs](https://github.com/jruby/activerecord-jdbc-adapter).

JRuby 9.2 supports Ruby 2.5 syntax, so only the Ruby 2.5 parts of `Appraisals` and `Rakefile` were modified.

#### _Note for reviewers_
_When reviewing the changes to `Rakefile`, I recommend to "Hide whitespace changes", as most changes are just indentation changes due to the removal of the `if RUBY_PLATFORM != 'java'` block._

### CODE REVIEW

File: lib/ddtrace/contrib/sequel/utils.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ module Sequel
       module Utils
         class << self
           def adapter_name(database)
-            Datadog::Utils::Database.normalize_vendor(database.adapter_scheme.to_s)
+            Datadog::Utils::Database.normalize_vendor(database.database_type.to_s)
```
Comment (ericmustin) [inline lib/ddtrace/contrib/sequel/utils.rb:8]: what's the motivation behind this change? will this change the behavior for current non jruby users?

**Comment by ericmustin** [comment]
@marcotc should we make an explicit note within the grpc and mysql2 integration docs that they are not supported on jruby?

File: lib/ddtrace/contrib/sequel/utils.rb (hunk 1 above)
Comment (marcotc) [reply lib/ddtrace/contrib/sequel/utils.rb:8]: `span.get_tag('sequel.db.vendor')` always returns `jdbc` by calling `database.adapter_scheme`, but returns the actual database adapter (e.g. `sqlite3`) by calling `database.database_type`.

MRI users won't see any change, as both methods return the same value under MRI. Also our MRI test suite is still ✅, so that's good :)

Comment (marcotc) [reply lib/ddtrace/contrib/sequel/utils.rb:8]: Ok, I look into the other adapters and I noticed an inconsistency with `mysql` (vs `mysql2`) with this change.
This brings out large underlying issues with our testing matrix for `sequel`: we only test it with `sqlite3` today.

I'll add some more testing for other adapters in this case to prevent such issues arising in the future.

Comment (marcotc) [reply lib/ddtrace/contrib/sequel/utils.rb:26]: I added `sequel` tests for MySQL and PostgreSQL (in addition to the original Sqlite one).
This should cover all combinations for the adapter name change.

The current changes pushed should now keep MRI builds intact but work properly with JDBC drivers from JRuby.

Comment (ericmustin) [reply lib/ddtrace/contrib/sequel/utils.rb:26]: ah gotcha, i think we're covered here 😅 .

**Comment by marcotc** [comment]
@ericmustin You make a good point, specially regarding changes introduced in https://github.com/DataDog/dd-trace-rb/pull/995 which were not publicly documented: I've added those to the documentation now.

> Also, just to be extra defensive, should we be explicit about versions we explicitly dont support?

I don't think that's necessary: we have always only supported versions that we explicitly mention as supported in our documentation.

File: spec/ddtrace/configuration/settings_spec.rb [hunk 2]
```diff
@@ -476,7 +476,7 @@
       subject(:rate_limit) { settings.sampling.rate_limit }
 
       context 'default' do
-        it { is_expected.to be 100 }
+        it { is_expected.to eq(100) }
```
Comment (marcotc) [inline spec/ddtrace/configuration/settings_spec.rb:479]: `be` performs identity comparison: same object.
`eq` performs equivalence comparison: same value.

There are many instances of `be(Numeric)` in our tests that were working because of Ruby MRI reuses the same object instance for small numbers, but JRuby doesn't in all cases.
